Sources/Extensions/SwiftStdlib/StringProtocolExtensions.swift

```diff
@@ -19,8 +19,33 @@ public extension StringProtocol where Index == String.Index {
     ///     - Parameter options: Options for the comparison.
     /// - Returns: The longest common suffix of the receiver and the given String
     public func commonSuffix<T: StringProtocol>(with aString: T, options: String.CompareOptions = []) -> String {
-        let reversedSuffix = String(reversed()).commonPrefix(with: String(aString.reversed()), options: options)
```

Comment:
You don't need to do this all by hand.
```Swift
public func commonSuffix<T: StringProtocol>(with aString: T, options: String.CompareOptions = []) -> String {
    var suffix = ""
    suffix.reserveCapacity(count)
    for (lhs, rhs) in zip(reversed(), aString.reversed()) {
        guard String(lhs).compare(String(rhs), options: options) == .orderedSame else { break }
        suffix.append(lhs)
    }
    return String(suffix.reversed())
}
```

Reply:
That makes sense to use zip since `Zip2Sequence` and `ReversedCollection` are "lazy".
But I think is not an issue handle the compare by hand since we avoid allocate new strings and you only handle options that make sense in this context of single character comparation (case Insentive and literal).  And maybe avoid reverse it at the end by insert(at: 0 ...) instead of the append on the suffix. 
Also I'm not sure about reserve capacity for count, maybe it make more sense to use `min(count, aString.count)`, but since String and StringProtocol don't conform to `RandomAccessCollection` the counts probably are not O(1), and also maybe we reserve space that we don't use e. g. Two large strings that don't have any common suffix. 
I'll try take a look at this later to check the options, thank's for the review  @guykogus  :))

Reply:
In the end, I think both implementations seem to have the same performance gain (avoid eager reverse). And using for with lazy reversed and Zip2Sequence to walk the strings by the index are just a simple implementation detail and more personal preference. So I think there's not a big problem to let this implementation here. :)) Let me know what you think.
cc @omaralbeik @SD10 Thoughts?

Reply:
I don't think the current implementation is correct. What if the caller of this function wanted to call with `.diacriticInsensitive`, for example? There's no need to check the options manually. I think that's the main benefit to my suggestion.
Also, I wouldn't suggest inserting at `0`/`startIndex` since appending to the beginning (assuming an array is used internally, which is pretty common practice) is O(n) while appending to the end (if you've reserved enough memory from the beginning) is O(1).

Reply:
I see the point, is just that it bothers me a bit have to convert the character to string to compare 🙃 and since I thought the only options that make sense character wise comparation were those two... but right I'll change :))